Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -45,6 +45,11 @@ internal final class DatadogCore {
     /// Telemetry monitor, if configured.
     var telemetry: Telemetry?
 
+    let readWriteQueue = DispatchQueue(
+        label: "com.datadoghq.ios-sdk-read-write",
+        target: .global(qos: .utility)
+    )
```

Comment:
There is one potential step we could look into in this PR. Now, by having an explicit access to this r/w queue we can simplify this test:

https://github.com/DataDog/dd-sdk-ios/blob/94819c05807fcd9b77bc85b8da0de59bc29ec82b/Tests/DatadogTests/Datadog/DatadogTests.swift#L445-L453

We `.sync {}` on each Feature's queue, but now it can be single `core.readWriteQueue.sync {}`.

Reply:
Good cartch!